Source/ASCollectionView.mm

```diff
@@ -333,6 +334,10 @@ - (void)dealloc
   ASPerformBackgroundDeallocation(&_rangeController);
 }
 
+- (BOOL)isDeallocating {
+  return _isDeallocating;
```

Comment:
This method can be called on multiple thread (i.e the editing queue of `ASDataController`). We should make sure it's thread-safe.

Reply:
If possible let's remove the isDeallocating method and instance variable itself, and just check the weak pointer for nil-ness.